### PR TITLE
fix(WebWorker): remove the platform-browser dependency on compiler

### PR DESCRIPTION
**Pull request by vicb**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit-message-format

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [x] Bugfix
- [x] Refactoring (no functional changes, no api changes)

**Does this PR introduce a breaking change?**
- [ ] Yes
- [x] No

/cc @IgorMinar

### CODE REVIEW

File: modules/@angular/integration_test/public_api_spec.ts [hunk 1]
```diff
@@ -410,11 +442,22 @@ var PLATFORM_BROWSER_DYNAMIC: string[] = [
   'CACHED_TEMPLATE_PROVIDER',
   'bootstrap',
   'UrlChangeEvent:dart',
-  'UrlChangeListener:dart'
+  'UrlChangeListener:dart',
+  'WORKER_APP_DYNAMIC_APPLICATION_PROVIDERS',
+  'WORKER_RENDER_DYNAMIC_APPLICATION_PROVIDERS',
+  'bootstrapApp',
+  'bootstrapRender',
```
Comment (IgorMinar) [inline modules/@angular/integration_test/public_api_spec.ts]: for consistency sake, we should call this `bootstrapWorkerApp` and `bootstrapWorkerRender`

File: modules/@angular/platform-browser/src/platform_browser.ts [hunk 2]
```diff
@@ -20,10 +12,10 @@ export {
   enableDebugTools,
   disableDebugTools,
   HAMMER_GESTURE_CONFIG,
-  HammerGestureConfig
+  HammerGestureConfig,
+  BROWSER_PLATFORM_MARKER
```
Comment (IgorMinar) [inline modules/@angular/platform-browser/src/platform_browser.ts]: is this right? I thought this was an implementation details

File: modules/@angular/platform-browser/src/web_workers/shared/messaging_api.ts [hunk 3]
```diff
@@ -3,6 +3,5 @@
  * You should not use these channels in your application code.
  */
 export const RENDERER_CHANNEL = "ng-Renderer";
-export const XHR_CHANNEL = "ng-XHR";
```
Comment (IgorMinar) [inline modules/@angular/platform-browser/src/web_workers/shared/messaging_api.ts]: why was this moved?

File: modules/@angular/platform-browser/src/web_workers/ui/router_providers.ts [hunk 4]
```diff
@@ -1,15 +1,24 @@
...
-import {APP_INITIALIZER, Provider, Injector, NgZone} from '@angular/core';
+import {BrowserPlatformLocation} from '../../browser/location/browser_platform_location';
+import {APP_INITIALIZER, Injector, NgZone} from '@angular/core';
 
-export const WORKER_RENDER_ROUTER = /*@ts2dart_const*/[
+/**
+ * A list of {@link Provider}s. To use the router in a Worker enabled application you must
+ * include these providers when setting up the render thread.
+ */
+export const WORKER_RENDER_ROUTER_PROVIDERS = /*@ts2dart_const*/[
```
Comment (IgorMinar) [inline modules/@angular/platform-browser/src/web_workers/ui/router_providers.ts]: why do we need special providers for the router? shouldn't we instead bind the BrowserPlatformLocation to a factory that would init the listeners and instantiate MessageBasedPlatformLocation ?

Comment (IgorMinar) [inline modules/@angular/platform-browser/src/web_workers/ui/router_providers.ts]: I see.. we need this because we don't have a way to lazily create an instance of the MessageBasedPlatformLocation from the worker side. so the workaround is that there are all these providers that you need to install on both sides and they are eagerly instantiated via initializers.

File: modules/@angular/platform-browser/src/webworker/worker_app.ts [hunk 5]
```diff
@@ -18,13 +15,20 @@ let _postMessage = {
...
   /* @ts2dart_Provider */ {provide: APP_INITIALIZER, useValue: setupWebWorker, multi: true}
 ];
 
+/**
+ * Use WORKER_APP_APPLICATION_PROVIDERS instead.
+ *
+ * @deprecated
+ */
+export const WORKER_APP_APPLICATION: Array<any /*Type | Provider | any[]*/> =
+    WORKER_APP_APPLICATION_PROVIDERS;
```
Comment (IgorMinar) [inline modules/@angular/platform-browser/src/webworker/worker_app.ts]: I would just document this change and not provide a deprecated api. the same with the other deprecation above. all this webworker stuff is quite experimental and we will still rename some of these apis anyway so this just adds noise.

File: modules/@angular/platform-browser/src/web_workers/worker/worker_adapter.dart [hunk 6]
```diff
@@ -0,0 +1,384 @@
+library angular2.dom.webWorkerAdapter;
+
+import 'abstract_html_adapter.dart';
+import 'package:angular2/platform/common_dom.dart';
+
```
Comment (IgorMinar) [inline modules/@angular/platform-browser/src/web_workers/worker/worker_adapter.dart]: please add a comment explaining what this file does just like you did in the TS version

File: modules/@angular/platform-browser/src/worker_app.ts [hunk 7]
```diff
@@ -27,20 +35,23 @@ export {
...
 export {PRIMITIVE} from './web_workers/shared/serializer';
 export * from './web_workers/shared/message_bus';
-export {WORKER_APP_ROUTER} from './web_workers/worker/router_providers';
+export {
+  WORKER_APP_ROUTER,
+  WORKER_APP_ROUTER_PROVIDERS
+} from './web_workers/worker/router_providers';
 
-export function workerAppPlatform(): PlatformRef {
+export function workerStaticAppPlatform(): PlatformRef {
```
Comment (IgorMinar) [inline modules/@angular/platform-browser/src/worker_app.ts]: why "static"? I think the platform is agnostic of the static/dynamic mode.

File: modules/@angular/platform-browser/src/worker_render.dart [hunk 8]
```diff
@@ -29,14 +29,14 @@ export '../src/web_workers/ui/router_providers.dart' show WORKER_RENDER_ROUTER;
 
 const WORKER_RENDER_APP = WORKER_RENDER_APPLICATION_COMMON;
 
-PlatformRef workerRenderPlatform() {
+PlatformRef workerStaticRenderPlatform() {
```
Comment (IgorMinar) [inline modules/@angular/platform-browser/src/worker_render.dart]: same as above - there is no "static" platform

File: modules/@angular/platform-browser/src/worker_render.dart [hunk 9]
```diff
@@ -29,14 +29,14 @@ export '../src/web_workers/ui/router_providers.dart' show WORKER_RENDER_ROUTER;
...
   if (isBlank(getPlatform())) {
     createPlatform(ReflectiveInjector.resolveAndCreate(WORKER_RENDER_PLATFORM));
   }
   return assertPlatform(WORKER_RENDER_PLATFORM_MARKER);
 }
 
-Future<ApplicationRef> bootstrapRender(
+Future<ApplicationRef> bootstrapStaticRender(
```
Comment (IgorMinar) [inline modules/@angular/platform-browser/src/worker_render.dart]: is there a difference between bootstrapStaticRender and bootstrapRender? I thought there was just one version

File: protractor-js-new-world.conf.js [hunk 10]
```diff
@@ -19,7 +19,6 @@ exports.config = {
     'dist/all/**/e2e_test/**/*_spec.js'
   ],
   exclude: [
-    '**/web_workers/**', // needs to be setup
     '**/key_events/**',  // can't tell why this is failing
```
Comment (IgorMinar) [inline protractor-js-new-world.conf.js]: I believe that you should be able to uncomment this one as well since the key events plugin is now exported

File: modules/@angular/platform-browser/src/platform_browser.ts (hunk 2 above)
Comment (vicb) [inline modules/@angular/platform-browser/src/platform_browser.ts]: should be moved to private exports

File: modules/@angular/platform-browser/src/web_workers/ui/router_providers.ts (hunk 4 above)
Comment (vicb) [inline modules/@angular/platform-browser/src/web_workers/ui/router_providers.ts]: rename to LOCATION

File: modules/@angular/platform-browser/src/worker_app.ts (hunk 7 above)
Comment (vicb) [inline modules/@angular/platform-browser/src/worker_app.ts]: merge -> static
